File: cplusplus/src/verify_transcript.cpp
#include "autograph/verify_transcript.h"

bool verify_transcript(const Chunk &transcript, const Chunk &their_identity_key,
                       const Chunk &their_secret_key, const Chunk &ciphertext) {
  try {
    Chunk signature = decrypt(their_secret_key, 0, ciphertext);
    bool verified = verify_signature(their_identity_key, transcript, signature);
    return verified;
  } catch (std::exception &error) {
    return false;
  }
}

File: typescript/src/verify-transcript.ts
import decrypt from './crypto/decrypt'
import verify from './crypto/verify'

const verifyTranscript = async (
  transcript: BufferSource,
  theirIdentityKey: BufferSource,
  theirSecretKey: BufferSource,
  ciphertext: BufferSource
) => {
  try {
    const signature = await decrypt(theirSecretKey, 0, ciphertext)
    const verified = await verify(transcript, theirIdentityKey, signature)
    return verified
  } catch (error) {
    return false
  }
}

export default verifyTranscript
